InterfaceParser/parsers/rpc_base.py

```diff
@@ -0,0 +1,782 @@
+
+        return Param(**params)
+
+    def _parse_function_param(self, element, prefix):
+        """Parse element as function parameter.
+
+        :param element: an instance of Element (from ElementTree)
+        :param prefix: empty string
+        :return: an instance of model.Param
+        """
+        params, subelements, attrib = self._parse_param_base_item(element, prefix)
+
+        default_value = None
+        default_value_string = self._extract_attrib(attrib, "defvalue")
+        if default_value_string is not None:
+            param_type = params["param_type"]
+            if isinstance(param_type, Boolean):
+                default_value = \
+                    self._get_bool_from_string(default_value_string)
+            elif isinstance(param_type, Integer):
+                try:
+                    default_value = int(default_value_string)
+                except:
+                    raise ParseError("Invalid value for integer: '{}'".format(default_value_string))
+            elif isinstance(param_type, Float):
+                try:
+                    default_value = float(default_value_string)
+                except:
+                    raise ParseError("Invalid value for float: '{}'".format(default_value_string))
+            elif isinstance(param_type, String):
+                default_value = default_value_string
+            elif isinstance(param_type, (Enum, EnumSubset)):
+                if isinstance(param_type, EnumSubset):
+                    allowed_elements = param_type.allowed_elements
+                else:
+                    allowed_elements = param_type.elements
+                if default_value_string not in allowed_elements:
+                    raise ParseError("Default value '{}' for parameter '{}' is not a member of {} '{}'"
+                                     .format(default_value_string, params["name"], type(param_type).__name__,
+                                             params["name"]))
+                default_value = allowed_elements[default_value_string]
+            else:
+                raise ParseError("Default value specified for " + type(param_type).__name__)
+        params["default_value"] = default_value
+
+        if len(attrib) != 0:
+            raise ParseError("Unexpected attributes in parameter '{}'".format(params["name"]))
+
+        if len(subelements) != 0:
+            raise ParseError("Unexpected subelements in parameter '{}'".format(params["name"]))
+
+        return Param(**params)
+
+    def _parse_param_base_item(self, element, prefix):
+        """Parse base param items.
+
+        :param element: an instance of Element (from ElementTree)
+        :param prefix: empty string
+        :return: params, other subelements and attributes.
+        """
+        params, subelements, attrib = self._parse_base_item(element, "")
+
+        since_version = self._extract_attrib(attrib, "since")
```

Comment:
This block could be placed into a method named _extract_all_base_attrib?

Reply:
there is no method "_extract_all_base_attrib" in rpc_base.py, and not clear what exactly you suggest to move to such new method? As well you sure that this change proposal right place for such refactoring?